ibm/mas_devops/roles/cp4d_wds/tasks/embeddedwds.yml

```diff
@@ -0,0 +1,310 @@
+  set_fact:
+    wds_instname: "discovery-assist-{{ mas_workspace_id }}"
+  when: mas_workspace_id is defined or mas_workspace_id != ""
+
+- name: "cp4d/wds: Print the Discovery Instance Name"
+  debug:
+    msg:
+      - "Discovery Instance Name.......... {{ wds_instname }}"
+
+- name: "cp4d/wds:Check if there's existing Discovery instance in CP4D"
+  when: wds_bearer_token is defined
+  uri:
+    url: "{{ cp4d_host_url }}/zen-data/v3/service_instances?fetch_all_instances=false&limit=9223372036854775807&offset=0"
+    validate_certs: no
+    method: GET
+    headers:
+      Authorization: "Bearer {{ wds_bearer_token }}"
+      cache-control: "no-cache"
+    status_code: 200,412
+    timeout: 30
+  register: query_instance_output
+
+- name: "cp4d/wds: find the WDS instance and query the wds uuid"
+  set_fact:
+    wds_number: "{{ item.id | int}}"
+    findwdsinst: true
+  with_items: "{{ query_instance_output.json.service_instances }}"
+  when:
+    - query_instance_output.status is defined and query_instance_output.status|int == 200
+    - item.addon_type == "discovery"
+    - item.display_name == wds_instname
+
+
+# 10. Provision the CP4D WDS instance and get its uuid
+# -----------------------------------------------------------------------------
+- name: "cp4d/wds: Create the WDS instance in CP4D when WDS instance not exist and query its uuid"
+  when: not ( findwdsinst is defined and findwdsinst )
+  block:
+    - name: create new WDS instance
+      uri:
+        url: "{{ cp4d_host_url }}/watson/common/discovery/api/ibmcloud/resource-controller/resource_instances"
+        validate_certs: no
+        method: POST
+        headers:
+          Authorization: "Bearer {{ wds_bearer_token }}"
+          cache-control: "no-cache"
+        body_format: "json"
+        body: "{{ lookup('template','templates/wds.json.j2') }}"
+        status_code: 200,412
+        timeout: 30
+      register: create_wds_output
+    - debug:
+        msg: "discovery instance ID.............. {{ create_wds_output.json.id }}"
+      when: create_wds_output.status|int == 200
+    - name: "cp4d/wds: Get the CP4D instance uuid number"
+      when: create_wds_output.status|int == 200
+      set_fact:
+        wds_number: "{{ create_wds_output.json.id | replace('-','')|int }}"
+
+# 11. Print Out the Discovery instance URL and crdential for use
+# -----------------------------------------------------------------------------
+- name: "cp4d/wds: Set and Print out Discovery instance info"
+  block:
+    - name: Get the CP4D instance url
+      set_fact:
+        wds_url: "{{ cp4d_host_url}}/discovery/{{ cpdnamespace }}-wd/instances/{{ wds_number }}/api"
+    - name: Print the WDS related info
+      debug:
+        msg:
+          - "WDS url ............................ {{ wds_url }}"
+          - "WDS Admin .......................... {{ wds_admin_username }}"
+          - "WDS Passw0rd ....................... {{ wds_admin_password }}"
```

Comment:
I don't think we want to output the password here

Reply:
removed